WoWsShipBuilder.ViewModels/Other/ShipComparisonViewModel.cs

```diff
@@ -0,0 +1,762 @@
+                    if (!shipList.Any(x => x.ShipDataContainer.AntiAirDataContainer is not null))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+
+                case ShipComparisonDataSections.Asw:
+                    if (!shipList.Any(x => x.ShipDataContainer.AswAirstrikeDataContainer is not null || x.ShipDataContainer.DepthChargeLauncherDataContainer is not null))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+
+                case ShipComparisonDataSections.AirStrike:
+                    if (!shipList.Any(x => x.ShipDataContainer.AirstrikeDataContainer is not null))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+
+                case ShipComparisonDataSections.RocketPlanes:
+                    if (!shipList.Any(x => x.RocketPlanesType.Any()))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+
+                case ShipComparisonDataSections.Rockets:
+                    if (!shipList.Any(x => x.RocketPlanesWeaponType.Any()))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+
+                case ShipComparisonDataSections.TorpedoBombers:
+                    if (!shipList.Any(x => x.TorpedoBombersType.Any()))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+
+                case ShipComparisonDataSections.AerialTorpedoes:
+                    if (!shipList.Any(x => x.TorpedoBombersWeaponType.Any()))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+
+                case ShipComparisonDataSections.Bombers:
+                    if (!shipList.Any(x => x.BombersType.Any()))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+
+                case ShipComparisonDataSections.Bombs:
+                    if (!shipList.Any(x => x.BombersWeaponType.Any()))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+
+                case ShipComparisonDataSections.Sonar:
+                    if (!shipList.Any(x => x.ShipDataContainer.PingerGunDataContainer is not null))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+            }
+        }
+
+        return dataSections;
+    }
+
+    public async Task ToggleDataSection(ShipComparisonDataSections dataSection)
```

Comment:
This is not a toggle, call it ChangeDataSection or something but not toggle. A Toggle can toggle between on/off or true/false